File: best Time to Buy and Sell Stock - E/sol.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int i = 0;
        int j = 1;
        int profit = 0;
        while (i < prices.size() && j < prices.size()) {
            if (prices[i] > prices[j]) {
                i = j;
            }
            else if (profit < prices[j] - prices[i]) {
                profit = prices[j] - prices[i];
            }
            j++;
        }
        return profit;
    }
};

File: best Time to Buy and Sell Stock - E/sol.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        i = 0
        j = 1
        profit = 0
        while i < len(prices) and j < len(prices):
            if prices[i] > prices[j]:
                i = j          
            elif prices[j] - prices[i] > profit:
                profit = prices[j] - prices[i]
            j += 1
        return profit
